src/epics/chatEpics.js

```diff
@@ -73,98 +67,34 @@ const suggestionEpic = (action$) =>
     .catch((err) => console.error(err));
 
 
-function createKeyPairObservable({ pincode = '' }) {
-  return Observable.create(function (observer) {
-    const urlHash = document.location.hash + pincode;
-    // 1. Get passphrase
-    const {
-      passphrase,
-      isNewPassphrase
-    } = getPassphrase(urlHash);
-
-    // 3. Get email based on passphrase
-    let email = getEmail(passphrase);
-    // 4. Decrypt to get key pair
-    miniLock.crypto.getKeyPair(passphrase, email, (keyPair) => {
-      miniLock.session.keys = keyPair;
-      miniLock.session.keyPairReady = true;
-      let mID = miniLock.crypto.getMiniLockID(keyPair.publicKey);
-      // 4. When we have keypair, login:
-
-      if (isNewPassphrase) {
-        document.location.hash = '#' + passphrase;
-      }
-
-      observer.next({ passphrase, keyPair, mID, isNewRoom: isNewPassphrase });
-      observer.complete();
-    });
-  });
-}
-
-function createDecryptMessageObservable({ message, mID, secretKey }) {
-  return Observable.create(function (observer) {
-    miniLock.crypto.decryptFile(message, mID, secretKey,
-      function (fileBlob, saveName, senderID) {
-        const reader = new FileReader();
-        reader.addEventListener("loadend", () => {
-          const authToken = reader.result;
-          observer.next(authToken);
-          observer.complete();
-        });
-
-        reader.readAsText(fileBlob);
-      });
+const initChatHandlerConnection = ({ authToken, secretKey, mID, isNewRoom }) => {
+  return chatHandler.initConnection({
+    authToken,
+    secretKey: secretKey,
+    mID
   });
```

Comment:
Step 7. Initialize the chat handler's connection. Since `chatHandler` is a shared global variable in this module, this is as far as we can go, without ripping all of the other observable stuff out of here too. This returns an observable.

Reply:
So much more readable! Bravo 👏